src/actions/run/pushAzureRepo.ts

```diff
@@ -96,19 +96,22 @@ export const pushAzureRepoAction = (options: {
       );
 
       const host = server ?? "dev.azure.com";
-      const integrationConfig = integrations.azure.byHost(host);
 
-      if (!integrationConfig) {
-        throw new InputError(
-          `No matching integration configuration for host ${host}, please check your integrations config`
-        );
-      }
+      const provider = DefaultAzureDevOpsCredentialsProvider.fromIntegrations(integrations);
 
-      if (!integrationConfig.config.token && !ctx.input.token) {
-        throw new InputError(`No token provided for Azure Integration ${host}`);
-      }
+      //TODO: we need the remoteUrl to get the credentials for a specific organization
```

Comment:
Does that mean that extra inputs will need to be added to the action to provide the host/org (or the full repo URL)?

Reply:
No, I have updated the `Git` wrapper from `@backstage/backend-common` to accept a function to handle authentication instead of static credentials. That function gets the `url` as input whenever the git library gets challenged for credentials. I made this PR before https://github.com/backstage/backstage/pull/20938 was merged, hence the `//TODO:`. 

I forked this plugin to our code base because I was getting no response on this PR. I have a working implementation now using the changes from https://github.com/backstage/backstage/pull/20938. I'll update this PR with my (local) changes so you can review and merge them.